### PR TITLE
Figure generation

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR is the first draft for figure 2. More edits to come in a separate PR. please let me know if you have any questions

### CODE REVIEW

File: figures/2.figure2/scripts/figure2.r [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+# aggregate the cytokine values across the treatments and cytokine
+cytokine_values_melted_agg <- cytokine_values_melted %>%
+    group_by(oneb_Metadata_Treatment_Dose_Inhibitor_Dose, Cytokine) %>%
+    summarise(Cytokine_Value = mean(Cytokine_Value))
+
+cytokine_values_melted_agg_filtered <- cytokine_values_melted_agg[cytokine_values_melted_agg$Cytokine %in% unique(tukey_results$cytokine),]
+head(cytokine_values_melted_agg_filtered)
+
+# un melt the data cytokine_values_melted_agg_filtered
```
Comment (jenna-tomkinson) [inline figures/2.figure2/scripts/figure2.r:241]:
```suggestion
# un melt the data cytokine_values_melted_agg_filtered
```
I would recommend giving a bit of a description of what being melted is or what it is to "un melt"

File: figures/2.figure2/scripts/figure2.r [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+# remove '[NSU]' from the column names
+colnames(cytokine_values_agg) <- gsub("\\[NSU\\]", "", colnames(cytokine_values_agg))
+
+
+cytokine_values_agg <- as.matrix(cytokine_values_agg)
+
+# set plot size
+options(repr.plot.width=25, repr.plot.height=5)
+heatmap_plot_all <- (
+  Heatmap(
```
Comment (jenna-tomkinson) [inline figures/2.figure2/scripts/figure2.r:206]: I am a bit confused here, this looks to be the same code in lines 256-271. Unless both are necessary, I would recommend to not repeat the same code.

File: figures/2.figure2/scripts/figure2.r [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+        row_names_gp = gpar(fontsize = 12),    # Row name label formatting
+        heatmap_legend_param = list(title = "Abundance", at = c(0, 1)),
+        # make the tiles rectangular
+        rect_gp = gpar(col = NA)
+    )
+)
+
+heatmap_anova_cytokines
+
+# read in the uamp results
```
Comment (jenna-tomkinson) [inline figures/2.figure2/scripts/figure2.r:275]:
```suggestion
# read in the umap results
```

File: figures/2.figure2/scripts/figure2.r [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+    )
+    + geom_point()
+)
+umap_plot_selected_treatments
+
+# show the plots first as assigned names
+# set plot size
+options(repr.plot.width=5, repr.plot.height=5)
+
+# cytokine_scatter_plot
```
Comment (jenna-tomkinson) [inline figures/2.figure2/scripts/figure2.r:320]: I would recommend removing the commented code from this line to line 328 unless they are used for something like titles but I am not sure what they are used for.

File: figures/2.figure2/scripts/figure2.r [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+# umap_plot_all_treatments
+# umap_plot_selected_treatments
+
+new_heatmap <- as.ggplot(heatmap_anova_cytokines)
+cytokine_scatter_plot_legend <- get_legend(cytokine_scatter_plot)
+umap_plot_selected_treatments_legend <- get_legend(umap_plot_selected_treatments)
+
+
+# remove legends
```
Comment (jenna-tomkinson) [inline figures/2.figure2/scripts/figure2.r:335]: Does this mean there is code in here to manually add the legend for the panel A plots? It looks like in the png that the legend is there for the UMAP.

**Review by jenna-tomkinson — CHANGES_REQUESTED** [review]
Great PR! This figure is a great first start! I am putting "request changes" since I am not sure if you plan on addressing all the figure comments in this PR or in another one. Please let me know what your plan is and I can change the review accordingly.

I have some comments to address regarding the code and figure, and I hope they will help with making the figure better!

File: figures/2.figure2/scripts/figure2.r (hunk 1 above)
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:241]: Addressed

File: figures/2.figure2/scripts/figure2.r [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+suppressPackageStartupMessages(suppressWarnings(library(patchwork)))
+suppressPackageStartupMessages(suppressWarnings(library(cowplot)))
+suppressPackageStartupMessages(suppressWarnings(library(RcppTOML)))
+suppressPackageStartupMessages(suppressWarnings(library(pheatmap)))
+suppressPackageStartupMessages(suppressWarnings(library(lattice)))
+suppressPackageStartupMessages(suppressWarnings(library("RColorBrewer")))
+suppressPackageStartupMessages(suppressWarnings(library(gplots)))
+suppressPackageStartupMessages(suppressWarnings(library(ComplexHeatmap)))
+suppressPackageStartupMessages(suppressWarnings(library(ggplotify)))
+# import the R them
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:12]: I am not sure yet either! I will remove

File: figures/2.figure2/scripts/figure2.r [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+
+# Figure 2A
+
+# Load data
+data_path_cytokine_values <- file.path("..","..","2.Nomic_nELISA_Analysis/0.Exploratory_Analysis/PBMC/results/PBMC_all_cytokine_values_per_treatment_per_well.csv")
+cytokine_values <- read.csv(data_path_cytokine_values, header = TRUE, sep = ",")
+# filter out the treatments that are not in the list
+cytokine_values <- cytokine_values[cytokine_values$oneb_Metadata_Treatment_Dose_Inhibitor_Dose %in% list_of_treatments,]
+
+list_of_treatments <- c(
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:29]: I will add comments to explain

File: figures/2.figure2/scripts/figure2.r [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+        data = cytokine_values,
+        aes(
+            x = IL.1.beta..NSU.,
+            y = CCL24..NSU.,
+            color = oneb_Metadata_Treatment_Dose_Inhibitor_Dose
+        )
+    )
+    + geom_point()
+    + xlab("IL-1 beta Abundance")
+    + ylab("TNF alpha Abundance")
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:84]: Addressed

File: figures/2.figure2/scripts/figure2.r (hunk 2 above)
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:206]: One is selected cytokines from anova tests and one is all cytokines

File: figures/2.figure2/scripts/figure2.r (hunk 4 above)
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:320]: Yes!

File: figures/2.figure2/scripts/figure2.r (hunk 5 above)
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:335]: Yes, absolutely

File: figures/2.figure2/scripts/figure2.r [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+
+
+# remove legends
+umap_plot_selected_treatments <- umap_plot_selected_treatments + theme(legend.position = "none")
+cytokine_scatter_plot <- cytokine_scatter_plot + theme(legend.position = "none")
+cytokine_scatter_plot1 <- cytokine_scatter_plot1 + theme(legend.position = "none")
+
+
+
+design <- "AB#CCD#
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply figures/2.figure2/scripts/figure2.r:342]: Yep, added!

File: figures/figure_themes.r [hunk 10]
```diff
@@ -0,0 +1,32 @@
+
+
+suppressPackageStartupMessages(library(ggplot2))
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply figures/figure_themes.r:3]: I think that is `nbconvert` doing the conversion

**Comment by MikeLippincott** [comment]
I would prefer to merge for now and circle back to address this pin. There might be major revisions in the future to this plot so I don't want to perfect it just yet... Thanks!

**Review by jenna-tomkinson — APPROVED** [review]
Since these will be addressed in another PR (that I would be happy to also review), I will approve this PR since the comments are resolved.

**Pull request merged**